### PR TITLE
chore: Better, conforming docstrings in SDK client.

### PR DESCRIPTION
**Pull request by wes-turner**
First step in standardizing our docstrings per the Google style guide. Mostly, this was just formatting. But in some cases, existing docstrings were incomplete or misleading. They have been fixed.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/experimental/client.py [hunk 1]
```diff
@@ -367,100 +354,114 @@ def get_models(
...
 def stream_trials_metrics(trial_ids: List[int], group: str) -> Iterable[TrialMetrics]:
-    """
-    Streams trial metrics for one or more trials sorted by
-    trial_id, trial_run_id and steps_completed.
+    """Iterate over the metrics for one or more trials.
+
+    This function collects TrialMetrics from a trial, sorted by `trial_id`, `trial_run_id`
+    and `steps_completed`.
+
+    .. warning::
```
Comment (wes-turner) [inline harness/determined/experimental/client.py:423]: Am I crazy? If streaming is going on, I sure don't understand how.

File: harness/determined/experimental/client.py [hunk 2]
```diff
@@ -177,70 +178,51 @@ def get_experiment(experiment_id: int) -> Experiment:
...
     return _determined.whoami()
 
 
 # DOES NOT REQUIRE SINGLETON (don't force a login in order to log out).
 def logout() -> None:
-    """
-    Get the :class:`~determined.experimental.client.User` representing the
-    current user.
-    """
+    """Log out of the current session."""
```
Comment (wes-turner) [inline harness/determined/experimental/client.py:225]: I'd like to document what this actually means.

File: harness/determined/experimental/client.py [hunk 3]
```diff
@@ -367,100 +354,114 @@ def get_models(
...
+    and `steps_completed`.
+
+    .. warning::
+       Contrary to its name, no streaming is actually done in this function. As more metrics are
+       computed on the master, they will not be appended to the iterator this function returns.
 
-    Arguments:
+    Args:
         trial_ids: List of trial IDs to get metrics for.
+        group: The metrics group to stream. Must either "training" or "validation".
```
Comment (wes-turner) [inline harness/determined/experimental/client.py:429]: > Must either

Must be either

**Review by azhou-determined — APPROVED** [review]
nice

**Pull request merged**